test/Test.agda

```diff
@@ -1,76 +1,144 @@
-
 module _ where
 
 open import Agda.Builtin.List
 open import Agda.Builtin.Nat
+-- open import Agda.Builtin.Float
+-- open import Agda.Builtin.Word
+-- open import Agda.Builtin.Char
 open import Agda.Builtin.Equality
 
 variable
   a b : Set
 
+-- ** Foreign HS code
+
+-- language extensions
 {-# FOREIGN AGDA2HS
 {-# LANGUAGE LambdaCase #-}
 #-}
 
+-- imports
 {-# FOREIGN AGDA2HS
 import Prelude hiding (map, sum, (++))
 import Data.Monoid
+import Data.Word
+#-}
+
+-- comments
+{-# FOREIGN AGDA2HS
+-- comment
+-- another comment
+bla :: Int -> Int
+bla n = n * 4
+
+{- multi
+   line
+   comment
+-}
 #-}
```

Comment:
There was a reason to put this between two `COMPILE` pragmas: to make sure that the sorting of `FOREIGN` blocks and `COMPILE`s is working.

Reply:
Oops, totally missed that, I will revert this.